### PR TITLE
Merge Resume global step checkpoint into master

### PR DESCRIPTION
**Pull request by fperezgamonal**
Merge changes to fix resuming from previous checkpoint. Basically:
* We fixed the global step not being properly resumed (training restarted at global_step = 0)
* This implied that the training schedule used the initial learning rate, weight decay, etc. instead of the ones of the resumed step.

This change however, implies that the training schedules that will be applied sequentially have to adapt the iterations at which a new step is taking. So, for instance, if we are fine-tuning with flyingthings3d, we start the count at 1.2M iterations and proceed (then step at 1.4M instead of simply writing 200K, etc.). Nevertheless the maximum number of iterations remains constant at 500K in this case.

### CODE REVIEW

**Comment by fperezgamonal** [comment]
*UPDATE*: added prints of current ram memory in usage to find leak!
